.github/BUILD.md

```diff
@@ -4,30 +4,96 @@
 
-- Are you about to create a pull request for Umbraco?
+- Are you about to [create a pull request for Umbraco][contribution guidelines]?
 - Are you trying to get to the bottom of a problem in your existing Umbraco installation?
 
 If the answer is yes, please read on. Otherwise, make sure to head on over [to the download page](https://our.umbraco.com/download) and start using Umbraco CMS as intended.
 
-**Table of contents**
+## Table of contents
 
-[Building from source](#building-from-source)
-  * [The quick build](#quick)
-  * [Build infrastructure](#build-infrastructure)
-    * [Properties](#properties)
-    * [GetUmbracoVersion](#getumbracoversion)
-    * [SetUmbracoVersion](#setumbracoversion)
-    * [Build](#build)
-    * [Build-UmbracoDocs](#build-umbracodocs)
-    * [Verify-NuGet](#verify-nuget)
-  * [Cleaning up](#cleaning-up)
+↖️ You can jump to any section by using the "table of contents" button ( ![Table of contents icon](img/tableofcontentsicon.svg) ) above.
 
-[Azure DevOps](#azure-devops)
 
-[Quirks](#quirks)
-  * [Powershell quirks](#powershell-quirks)
-  * [Git quirks](#git-quirks)
+## Debugging source locally
 
+Did you read ["Are you sure"](#are-you-sure)?
+
+[More details about contributing to Umbraco and how to use the GitHub tooling can be found in our guide to contributing.][contribution guidelines]
+
+If you want to run a build without debugging, see [Building from source](#building-from-source) below. This runs the build in the same way it is run on our build servers.
+
+#### Debugging with VS Code
+
+In order to build the Umbraco source code locally with Visual Studio Code, first make sure you have the following installed.
+
+  * [Visual Studio Code](https://code.visualstudio.com/)
+  * [dotnet SDK v5.0](https://dotnet.microsoft.com/en-us/download)
+  * [Node.js v10+](https://nodejs.org/en/download/)
+  * npm v6.4.1+ (installed with Node.js)
+  * [Git command line](https://git-scm.com/download/)
+
+Open the root folder of the repository in Code.
+
+To build the front end you'll need to open the command pallet (<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>) and run `>Tasks: Run Task` followed by `Client Install` and then run the `Client Build` task in the same way.
+
+You can also run the tasks manually on the command line:
+
+```
+cd src\Umbraco.Web.UI.Client
+npm install
+npm run watch
+```
+
+or
+
+```
+cd src\Umbraco.Web.UI.Client
+npm install
+gulp watch
+```
+
+**The initial Gulp build might take a long time - don't worry, this will be faster on subsequent runs.**
+
+You might run into [Gulp quirks](#gulp-quirks).
+
+The caching for the back office has been described as 'aggressive' so we often find it's best when making back office changes to disable caching in the browser to help you to see the changes you're making.
```

Comment:
Is it worth us adding some links to instructions on how to disable caching for the major browers?

Reply:
That's a good shout. I wonder if I can find a link to an explainer elsewhere, to save us maintaining those instructions.